src/games/strategy/triplea/ai/fastAI/FastOddsEstimator.java

```diff
@@ -0,0 +1,146 @@
+package games.strategy.triplea.ai.fastAI;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.oddsCalculator.ta.AggregateResults;
+import games.strategy.triplea.oddsCalculator.ta.IOddsCalculator;
+import games.strategy.triplea.oddsCalculator.ta.OddsCalculatorListener;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class FastOddsEstimator implements IOddsCalculator {
+
+  private GameData data;
+  private PlayerID attacker = null;
+  private PlayerID defender = null;
+  private Territory location = null;
+  private Collection<Unit> attackingUnits = new ArrayList<Unit>();
+  private Collection<Unit> defendingUnits = new ArrayList<Unit>();
+  private Collection<Unit> bombardingUnits = new ArrayList<Unit>();
+  private Collection<TerritoryEffect> territoryEffects = new ArrayList<TerritoryEffect>();
+
+  @Override
+  public void setGameData(final GameData data) {
+    this.data = data;
+  }
+
+  @Override
+  public void setCalculateData(final PlayerID attacker, final PlayerID defender, final Territory location,
+      final Collection<Unit> attackingUnits, final Collection<Unit> defendingUnits,
+      final Collection<Unit> bombardingUnits,
+      final Collection<TerritoryEffect> territoryEffects, final int runCount) {
+    this.attacker = attacker;
+    this.defender = defender;
+    this.location = location;
+    this.attackingUnits = attackingUnits;
+    this.defendingUnits = defendingUnits;
+    this.bombardingUnits = bombardingUnits;
+    this.territoryEffects = territoryEffects;
+  }
+
+  @Override
+  public AggregateResults calculate() {
+    final double winPercentage = ProBattleUtils.estimateStrengthDifference(location,
```

Comment:
I don't think that is reason to allow the code to drift/degrade. Why make a problem worse leaving more work for yourself?